src/distribution/mod.rs

```diff
@@ -269,3 +273,24 @@ pub trait CheckedDiscrete<T, K> {
     /// ```
     fn checked_ln_pmf(&self, x: T) -> Result<K>;
 }
+
+/// The 'IdealSoliton' trait provides an interface for interacting
+/// with discrete statistical distributions from integers 1..N with
+/// N as the single parameter for the distribution
+pub trait Soliton<T, K> {
```

Comment:
Traits are often, but not always, a contract on types one does not yet know about. In this case the trait is superfluous, as there only seem to be two soliton distributions, both of which are provided.